File: ios-app/FiboHealth/Models/SessionState.swift
import Foundation

struct SessionState: Codable {
    var caloriesConsumed: Double
    var caloriesRemaining: Double
    var lastScanFood: String
    var lastScanKcal: Double

    enum CodingKeys: String, CodingKey {
        case caloriesConsumed  = "calories_consumed"
        case caloriesRemaining = "calories_remaining"
        case lastScanFood      = "last_scan_food"
        case lastScanKcal      = "last_scan_kcal"
    }

    static var empty: SessionState {
        SessionState(caloriesConsumed: 0, caloriesRemaining: 0,
                     lastScanFood: "", lastScanKcal: 0)
    }
}

File: android-app/app/src/main/java/com/fibonacci/fibohealth/data/model/SessionState.kt
package com.fibonacci.fibohealth.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SessionState(
    @SerialName("calories_consumed")  val caloriesConsumed: Float = 0f,
    @SerialName("calories_remaining") val caloriesRemaining: Float = 0f,
    @SerialName("last_scan_food")     val lastScanFood: String = "",
    @SerialName("last_scan_kcal")     val lastScanKcal: Float = 0f
)
